src/main/java/de/zalando/aruha/nakadi/service/subscription/StreamingContext.java

```diff
@@ -78,61 +87,52 @@ public long getKafkaPollTimeout() {
 
     @Override
     public void stream() throws InterruptedException {
-        currentState = new State() {
-            @Override
-            public void onEnter() {
-            }
-        };
+        streamInternal(new StartingState());
+    }
 
+    void streamInternal(final State firstState) throws InterruptedException {
         // Add first task - switch to starting state.
-        switchState(new StartingState());
+        switchState(firstState);
 
-        while (null != currentState) {
-            final Runnable task = taskQueue.poll(1, TimeUnit.MINUTES);
+        while (currentState != DEAD_STATE) {
+            // Wait forever
```

Comment:
I don't think that it's a big problem to wake up once in hour.
Actually there is no 'real forever' here possible, 1 hour is relatively big compared to commit timeout, batch_timeout and stream_timeout